Harvest.Core/Domain/Invoice.cs

```diff
@@ -35,6 +41,12 @@ internal static void OnModelCreating(ModelBuilder modelBuilder)
                 .WithMany(p => p.Expenses)
                 .HasForeignKey(a => a.InvoiceId)
                 .OnDelete(DeleteBehavior.Restrict);
+
+            modelBuilder.Entity<Transfer>()
+                .HasOne(a => a.Invoice)
+                .WithMany(p => p.Transfers)
+                .HasForeignKey(a => a.InvoiceId)
+                .OnDelete(DeleteBehavior.Restrict);
```

Comment:
I've seen this in a lot of our code. I figured it was to force developers to consider all relationships that will be impacted and to be explicit when deleting.

Reply:
Yea but specially what it does is to set the FK to `null` of the sub entity.  So, in this case if you delete the `Invoice` it would set `Transfer.InvoiceId = null` of any transfers which had this invoice ID.  Since the invoiceId is required that's just an FK error.